conformance_tests/tools/metrics/src/test_metric.cpp

```diff
@@ -1321,6 +1324,197 @@ TEST_F(
+                                    nullptr);
+    lzt::close_command_list(commandList);
+
+    for (auto groupInfo : metricGroupInfo) {
+      LOG_INFO << "test metricGroup name " << groupInfo.metricGroupName;
+      lzt::activate_metric_groups(device, 1, &groupInfo.metricGroupHandle);
+
+      ze_event_handle_t eventHandle;
+      lzt::zeEventPool eventPool;
+      eventPool.create_event(eventHandle, ZE_EVENT_SCOPE_FLAG_HOST,
+                              ZE_EVENT_SCOPE_FLAG_HOST);      
+
+      auto startTime = std::chrono::system_clock::now();
+      zet_metric_streamer_handle_t metricStreamerHandle =
+          lzt::metric_streamer_open_for_device(
+              device, groupInfo.metricGroupHandle, eventHandle,
+              notifyEveryNReports, samplingPeriod);
+      ASSERT_NE(nullptr, metricStreamerHandle);
+
+      lzt::execute_command_lists(commandQueue, 1, &commandList, nullptr);
+      lzt::synchronize(commandQueue, std::numeric_limits<uint64_t>::max());
+
+      double minimumTimeBeforeEventIsExpected = notifyEveryNReports * (static_cast<double>(samplingPeriod) / 1000000000);
+      // Initializing a 5% error buffer to prevent corner cases 
+      double errorBuffer = 0.05 * minimumTimeBeforeEventIsExpected;
+      LOG_INFO << "minimumTimeBeforeEventIsExpected " << minimumTimeBeforeEventIsExpected;
+      LOG_INFO << "errorBuffer " << errorBuffer;
+      
+      ze_result_t eventResult;
+      while (true) {
+        auto currentTime = std::chrono::system_clock::now();
+        std::chrono::duration<double> elapsedSeconds = currentTime - startTime;
+        if (elapsedSeconds.count() >= (minimumTimeBeforeEventIsExpected - errorBuffer)) {
+          break;
+        }
+        eventResult = zeEventQueryStatus(eventHandle);
+        EXPECT_EQ(eventResult, ZE_RESULT_NOT_READY);
+        std::this_thread::sleep_for(std::chrono::milliseconds(1000));
+      }
+
+      // Sleep again for 2x the buffer time to ensure corner cases are avoided. 2x is the amount to maintain the timeline itself.
+      uint32_t sleep = std::ceil(2 * errorBuffer);
+      LOG_INFO << "additional sleep before expecting event to be ready " << sleep;
+      std::this_thread::sleep_for(std::chrono::seconds(sleep));
+
+      eventResult = zeEventQueryStatus(eventHandle);
+      EXPECT_EQ(eventResult, ZE_RESULT_SUCCESS);
+
+      lzt::metric_streamer_close(metricStreamerHandle);
+      lzt::deactivate_metric_groups(device);
+      eventPool.destroy_event(eventHandle);     
+    }
+    lzt::destroy_function(function);
+    lzt::free_memory(a_buffer);
+    lzt::free_memory(b_buffer);
+    lzt::free_memory(c_buffer);
+    lzt::reset_command_list(commandList);
+    lzt::destroy_command_queue(commandQueue);
+    lzt::destroy_command_list(commandList);
+  }
+}
+
+TEST_F(
+    zetMetricStreamerTest,
+    GivenValidMetricGroupWhenTimerBasedStreamerIsCreatedThenExpectStreamerToNotifyEventAtProperTimeAndCorrectNumberOfReportsAreGenerated) {
```

Comment:
please elaborate a bit on how the feature coverage of this test is different from the one above. From my review it seems to b testing the same thing but instead of looping it just has a single wait based on predicted time.

Reply:
This test is trying to check if we have the correct number of reports generated after the even was generated. Previous test does not do the read and focuses only on proper even generation.

Reply:
but, can you add the sanity check of events generated to the test using the loop to check the event ?  I'm trying to avoid very similar test cases that we later need to maintain

Reply:
resolved in [comment ](https://github.com/oneapi-src/level-zero-tests/pull/107#discussion_r1848190530)